### PR TITLE
make-deploy example: run stacks on docker-machines

### PR DESCRIPTION
**Pull request by parente**
@jtyberg and I spent the morning hacking on this KISS automation for deploying singular docker-stack images to docker-machine controlled hosts. The included README should say it all.

This PR comes in response to various requests about how to go beyond "Great, I can run a jupyter in isolated in docker on my local machine." It's definitely not the end-all, be-all (it's based on Makefiles!), but we think it's a reasonably simple, raw example from which people can learn.

### CODE REVIEW

**Comment by parente** [comment]
Now has a Let's Encrypt recipe.

**Comment by parente** [comment]
I don't want to merge this one myself without another set of eyes on it with regards to whether it belongs in docker-stacks or not, whether we've gone totally crazy with make, whether the README even makes sense, etc.

@rgbkrk care to comment? From the makefiles / README can you figure out how to use it against Carina? (Seems like it would be really, really simple to do so.)

File: examples/make-deploy/README.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+### How do I run over HTTPS?
+
+Instead of `make notebook`, run `make self-signed-notebook PASSWORD=your_desired_password`. This target gives you a notebook wtih a self-signed certificate.
+
+### That self-signed certificate is a pain. Let's Encrypt?
+
+Yes. Please.
+
+```
+make letsencrypt FQDN=host.mydomain.com EMAIL=myemail@somewhere.com
```
Comment (rgbkrk) [inline examples/make-deploy/README.md]: Nice!

**Comment by rgbkrk** [comment]
Yeah, it wouldn't be too bad to set it up for `carina`. While Rackspace is part of docker-machine, Carina is not (differences in how certs are used/setup). It comes down to `carina create <name>` followed by `eval "$( carina env <name> )"`. Then it's Docker the rest of the way.

File: examples/make-deploy/softlayer.makefile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+softlayer-vm: export SOFTLAYER_CPUS?=4
```
Comment (jtyberg) [inline examples/make-deploy/softlayer.makefile]: I just tried creating a SoftLayer VM.  The SOFTLAYER_CPU variable is singular, not plural.

**Comment by parente** [comment]
@rgbkrk 

> While Rackspace is part of docker-machine, Carina is not (differences in how certs are used/setup). It comes down to carina create <name> followed by eval "$( carina env <name> )". Then it's Docker the rest of the way.

I imagine the Carina bits could be a another carina.makefile that just uses the Carina CLI to do whatever is necessary to set the local env vars for the rest of the docker targets to "just work."

**Comment by parente** [comment]
So my main remaining question is where does this make hack belong? In this repo? Somewhere else? It largely depends on how much we're willing to support it. My stance is that it's an example at best that folks can use as-is or change as they see fit.

**Comment by rgbkrk** [comment]
> I imagine the Carina bits could be a another carina.makefile that just uses the Carina CLI to do whatever is necessary to set the local env vars for the rest of the docker targets to "just work."

Sounds good to me.

**Comment by rgbkrk** [comment]
These stand pretty well in examples, I think this is fine.

**Pull request merged**